lib/reconcile/astarte_dashboard.go

```diff
@@ -181,11 +193,16 @@ func getAstarteDashboardConfigMapData(cr *apiv1alpha1.Astarte, dashboard apiv1al
 	} else {
 		dashboardConfig["appengine_api_url"] = dashboard.Config.AppEngineAPIURL
 	}
-	if dashboard.Config.FlowAPIURL == "" {
+	if dashboard.Config.FlowAPIURL == "" && !isAstarte10 {
 		dashboardConfig["flow_api_url"] = getBaseAstarteAPIURL(cr) + "/flow/"
 	} else {
```

Comment:
These else blocks don't really make a lot of sense. In case you're dealing with Astarte 1.0+, the config will contain an empty `flow_api_url` field. You probably wanted to run this else only if the explicit URL isn't blank. Also, I'm not really keen on how you do this selectively, whereas you just go ahead and always set `astarte_api_url` regardless. I guess I'm missing some pieces of the puzzle?